app/schema/item.py

```diff
@@ -17,3 +15,48 @@ def construct_state(self):
 
     def get_state_class(self):
         pass
+
+    def validate(self, state):
+        if isinstance(state, self.get_state_class()):
+            is_valid = True
+            for child_state in state.children:
+                child_schema = self.questionnaire.get_item_by_id(child_state.id)
+                child_valid = child_schema.validate(child_state)
```

Comment:
Wasn't sure I understood where you were coming from... The reason I've done it this way is to allow `is_valid` to represent the state of the current item, while the method itself interrogates all children.  If any of the children are in valid, then the current_item is invalid, but we presume it to be valid until a child fails.